api/src/middleware/publications.js

```diff
@@ -53,6 +53,21 @@ const createPublicationValidation = [
       .if(body("yearPublished")
       .exists())
       .isLength(4),
+    body("category.type", "Error: Category type must not be empty.")
+      .notEmpty(),
+    body("category.type", "Error: Category type does not match any of [\"CONFERENCE\", \"JOURNAL\"].")
+      .if(body("category.type")
+      .exists()
+      .notEmpty())
+      .isIn(["CONFERENCE", "JOURNAL"]),
```

Comment:
I'm not too sure how this validation works but when we write:
```javascript
body("category.type", "Error: Category type must not be empty.")
      .notEmpty(),
    body("category.type", "Error: Category type does not match any of [\"CONFERENCE\", \"JOURNAL\"].")
      .if(body("category.type")
      .exists()
      .notEmpty())
      .isIn(["CONFERENCE", "JOURNAL"]),
      ```
     isn't the second `notEmpty()` check and `exists()` redundant?

Reply:
The first one returns error when the category type is an emtpy string or not provided while the second one checks whether category type is one of the "CONFERENCE" or "JOURNAL" given the category type is provided.